pkg/scheduler/plugins/drf/drf.go

```diff
@@ -83,18 +89,66 @@ func (drf *drfPlugin) OnSessionOpen(ssn *framework.Session) {
 		drf.updateShare(attr)
 
 		drf.jobAttrs[job.UID] = attr
+
+		if namespaceOrderEnabled {
+			nsOpts, found := drf.namespaceOpts[job.Namespace]
+			if !found {
+				nsOpts = &drfAttr{
+					allocated: api.EmptyResource(),
+				}
+				drf.namespaceOpts[job.Namespace] = nsOpts
+			}
+			// all task in job should have the same namespace with job
+			nsOpts.allocated.Add(attr.allocated)
+			drf.updateShare(nsOpts)
+		}
 	}
 
 	preemptableFn := func(preemptor *api.TaskInfo, preemptees []*api.TaskInfo) []*api.TaskInfo {
 		var victims []*api.TaskInfo
 
+		addVictim := func(candidate *api.TaskInfo) {
+			victims = append(victims, candidate)
+		}
+
 		latt := drf.jobAttrs[preemptor.Job]
 		lalloc := latt.allocated.Clone().Add(preemptor.Resreq)
 		_, ls := drf.calculateShare(lalloc, drf.totalResource)
 
+		var lNsShareWeighted float64
+		if namespaceOrderEnabled {
+			lWeight := ssn.NamespaceInfo[preemptor.Namespace].GetWeight()
+			lNsAtt := drf.namespaceOpts[preemptor.Namespace]
+			lNsAlloc := lNsAtt.allocated.Clone().Add(preemptor.Resreq)
+			_, lNsShare := drf.calculateShare(lNsAlloc, drf.totalResource)
+			lNsShareWeighted = lNsShare / float64(lWeight)
+		}
+
 		allocations := map[api.JobID]*api.Resource{}
+		namespaceAllocation := map[string]*api.Resource{}
 
 		for _, preemptee := range preemptees {
+			if namespaceOrderEnabled && preemptor.Namespace != preemptee.Namespace {
```

Comment:
more comments

Reply:
Better refactor this part. since the change is little too hardcoded.

Reply:
refactor this part, namespace order policy is now a separate code block out of job order policy.